### PR TITLE
Feature/better receivers

### PR DESCRIPTION
**Pull request by Barisimre**
- Chip when the pass fails x amount of times
- If close to goal, shoot quicker
- Offensive positioning closer to goal

### CODE REVIEW

File: roboteam_ai/src/conditions/HasClearShot.cpp [hunk 1]
```diff
@@ -17,12 +17,17 @@ HasClearShot::HasClearShot(std::string name, bt::Blackboard::Ptr blackboard)
...
 HasClearShot::Status HasClearShot::onUpdate() {
+    MIN_VIEW_AT_GOAL = 0.2;
 
 	// return failure if the robot is too far away for a shot at goal
     if ((Vector2(ball->pos) - world::field->get_their_goal_center()).length() > MAX_SHOOTING_DISTANCE) {
         return Status::Failure;
     }
 
+    if (world::field->pointIsInDefenceArea(robot->pos, false, 0.5, false)) {
+        MIN_VIEW_AT_GOAL /= 4;
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/conditions/HasClearShot.cpp]: capital constant? but not constant? wth

File: roboteam_ai/src/control/positionControllers/ForcePosControl.h [hunk 2]
```diff
@@ -24,7 +24,7 @@ class ForcePosControl : public PosController {
 private:
     const double FORCE_WEIGHT_US = 1.0;
     const double FORCE_WEIGHT_THEM = 2.0;
-    const double FORCE_WEIGHT_BALL = 1.0;
+    const double FORCE_WEIGHT_BALL = 2.0;
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/control/positionControllers/ForcePosControl.h]: c

Comment (ThijsLuttikhuis) [reply roboteam_ai/src/control/positionControllers/ForcePosControl.h]: 2.99792458e8ms-1

File: roboteam_ai/src/control/shotControllers/ShotController.cpp [hunk 3]
```diff
@@ -39,6 +39,7 @@ ShotData ShotController::getShotData(world::Robot robot, Vector2 shotTarget, boo
        if (hasBall && !genevaIsTurning) {
            shotData = shoot(robot, shotTarget, chip);
        } else if (hasBall && genevaIsTurning) {
+            backDown(robot, currentDesiredGeneva);
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/control/shotControllers/ShotController.cpp]: why would you move slower? just go to the ball, contest it and wait for geneva instead.
that would be my idea.
do with it what you want <3 me

File: roboteam_ai/src/skills/Attack.cpp [hunk 4]
```diff
@@ -18,17 +25,12 @@ Attack::Attack(string name, bt::Blackboard::Ptr blackboard)
 }
 
 void Attack::onInitialize() {
-    shot = false;
-    shotControl = std::make_shared<control::ShotController>();
+    shotControl = std::make_shared<control::ShotController>(control::ShotPrecision::LOW, control::BallSpeed::MAX_SPEED, true);
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/skills/Attack.cpp]: does not need to be a pointer but >> it's okay << what rolf would say

File: roboteam_ai/src/skills/Pass.cpp [hunk 5]
```diff
@@ -56,19 +64,30 @@ Pass::Status Pass::onUpdate() {
...
-        if (shot && ballIsMovingFast) {
+        if (hasShot && ballIsMovingFast) {
             coach::g_pass.setPassed(true);
             return Status::Success;
         }
 
-        if(!shot && !control::ControlUtils::clearLine(ball->pos, robotToPassTo->pos, world::world->getWorld(), 1)) {
-            return Status::Failure;
+        if(!chip && !hasShot && !control::ControlUtils::clearLine(ball->pos, robotToPassTo->pos, world::world->getWorld(), 1)) {
+            if (failsUntilChip == -1) {
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/skills/Pass.cpp]: comment on what it does plox

File: roboteam_ai/src/conditions/HasClearShot.cpp (hunk 1 above)
Comment (Barisimre) [reply roboteam_ai/src/conditions/HasClearShot.cpp]: done

File: roboteam_ai/src/control/shotControllers/ShotController.cpp (hunk 3 above)
Comment (Barisimre) [reply roboteam_ai/src/control/shotControllers/ShotController.cpp]: Ah this is fine, it's being refactored anyways

File: roboteam_ai/src/skills/Attack.cpp (hunk 4 above)
Comment (Barisimre) [reply roboteam_ai/src/skills/Attack.cpp]: Nice

File: roboteam_ai/src/skills/Pass.cpp (hunk 5 above)
Comment (Barisimre) [reply roboteam_ai/src/skills/Pass.cpp]: ok

**Pull request merged**